kuma/wiki/apps.py

```diff
@@ -93,12 +93,16 @@ def on_document_save(self, sender, instance, **kwargs):
           cache for the given document
         - trigger the cache invalidation of the contributor bar for the given
           document
+        - trigger the renewal of the code sample job generation
         """
         async = kwargs.get('async', True)
         invalidate_zone_urls_cache(instance, async=async)
         invalidate_zone_stack_cache(instance, async=async)
+
         DocumentContributorsJob().invalidate(instance.pk)
 
+        DocumentCodeSampleJob(generation_args=[instance.pk]).renew()
```

Comment:
*(moving from commit-level comment)*

This will cause a new cryptographically strong random number to be generated, and stored in memcached for a very long time, on **every** page save, not just those with code sample source.  Instead, it should be something like:

```
cache.delete(DocumentCodeSampleJob(generation_args=[instance.pk]).generation_key)
```
which should probably be a helper method on ``GenerationJob``.  This will delete any existing generation key, causing invalidation and ``.renew()`` on use.